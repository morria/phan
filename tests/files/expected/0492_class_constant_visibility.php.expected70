%s:23 PhanPropertyAccessSignatureMismatch Access level to protected $prop must be compatible with public $prop defined in %s:4
%s:25 PhanAccessNonStaticToStaticProperty Cannot make non static property \X->instance_prop into the static property \Y::$instance_prop
%s:26 PhanAccessStaticToNonStaticProperty Cannot make static property \X::$static_prop into the non static property \Y->static_prop
%s:28 PhanAccessSignatureMismatch Access level to function test() must be compatible with function test() defined in %s:7
