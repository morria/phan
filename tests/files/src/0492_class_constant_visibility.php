<?php

class X {
    public $prop;
    public $instance_prop;
    public static $static_prop;
    public function test() {}

    private const pri1 = 11;
    private const pri2 = 12;
    private const pri3 = 13;

    protected const pro1 = 21;
    protected const pro2 = 22;
    protected const pro3 = 23;

    public const pub1    = 31;
    public const pub2    = 32;
    public const pub3    = 33;
}

class Y extends X {
    protected $prop;

    public static $instance_prop;
    public $static_prop;

    protected function test() {}

    private const pri1   = 111;
    protected const pri2 = 112;
    public const pri3    = 113;

    private const pro1   = 121;
    protected const pro2 = 122;
    public const pro3    = 123;

    private const pub1   = 131;
    protected const pub2 = 132;
    public const pub3    = 133;
}
var_export(Y::pub2);
