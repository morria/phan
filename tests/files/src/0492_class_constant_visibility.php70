<?php

class X {
    public $prop;
    public $instance_prop;
    public static $static_prop;
    public function test() {}












}

class Y extends X {
    protected $prop;

    public static $instance_prop;
    public $static_prop;

    protected function test() {}












}
