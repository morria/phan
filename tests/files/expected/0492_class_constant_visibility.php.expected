%s:23 PhanPropertyAccessSignatureMismatch Access level to protected $prop must be compatible with public $prop defined in %s:4
%s:25 PhanAccessNonStaticToStaticProperty Cannot make non static property \X->instance_prop into the static property \Y::$instance_prop
%s:26 PhanAccessStaticToNonStaticProperty Cannot make static property \X::$static_prop into the non static property \Y->static_prop
%s:28 PhanAccessSignatureMismatch Access level to function test() must be compatible with function test() defined in %s:7
%s:34 PhanConstantAccessSignatureMismatch Access level to private \Y::pro1 must be compatible with protected \X::pro1 defined in %s:13
%s:38 PhanConstantAccessSignatureMismatch Access level to private \Y::pub1 must be compatible with public \X::pub1 defined in %s:17
%s:39 PhanConstantAccessSignatureMismatch Access level to protected \Y::pub2 must be compatible with public \X::pub2 defined in %s:18
%s:42 PhanAccessClassConstantProtected Cannot access protected class constant \Y::pub2 defined at %s:39
